devtools/common/src/main/java/io/quarkus/cli/commands/AddExtensions.java

```diff
@@ -24,44 +26,64 @@ public AddExtensions(final File pom) throws IOException {
-                        return hasTag || machName;
-                    })
-                    .findAny();
-
-            if (optional.isPresent()) {
-                final Extension extension = optional.get();
-
-                if (!MojoUtils.hasDependency(model, extension.getGroupId(), extension.getArtifactId())) {
-                    System.out.println("Adding extension " + extension.managementKey());
-                    model.addDependency(extension
-                            .toDependency(containsBOM(model) &&
-                                    isDefinedInBom(dependenciesFromBom, extension)));
-                    updated = true;
+            Dependency res;
+            if (!dependency.contains(":")) {
+                String groupId, artifactId;
+                groupId = QUARKUS_GROUPID;
+                if (!dependency.startsWith(EXTENSION_ARTIFACTID_PREFIX)) {
+                    artifactId = EXTENSION_ARTIFACTID_PREFIX + dependency.trim().toLowerCase();
                 } else {
-                    System.out.println("Skipping extension " + extension.managementKey() + ": already present");
+                    artifactId = dependency.trim().toLowerCase();
                 }
-            } else if (dependency.contains(":")) {
-                Dependency parsed = MojoUtils.parse(dependency);
-                System.out.println("Adding dependency " + parsed.getManagementKey());
-                model.addDependency(parsed);
+                res = new Dependency();
+                res.setGroupId(groupId);
+                res.setArtifactId(artifactId);
+            } else {
+                if (dependency.startsWith(QUARKUS_GROUPID)) {
+                    dependency = dependency.toLowerCase();
+                }
+                res = MojoUtils.parse(dependency.trim());
+            }
+
+            if (QUARKUS_GROUPID.equals(res.getGroupId())) {
+                Optional<Extension> optional = loadedExtensions.stream()
+                        .filter(d -> {
+                            return d.getArtifactId()
+                                    .equals(res.getArtifactId());
+                        })
+                        .findAny();
+
+                if (!optional.isPresent()) {
+                    throw new CommandsException("Cannot find a Quarkus extension matching " + dependency);
+                } else if ((res.getVersion() == null || res.getVersion().isEmpty()) && !containsBOM(model)
+                        && isDefinedInBom(dependenciesFromBom, res)) {
+                    res.setVersion(optional.get().getVersion());
+                }
+            }
+
+            if (!MojoUtils.hasDependency(model, res.getGroupId(), res.getArtifactId())) {
+                if (res.getVersion() != null && res.getVersion().isEmpty() && containsBOM(model) &&
+                        isDefinedInBom(dependenciesFromBom, res)) {
+                    res.setVersion(null);
```

Comment:
...and reset it to null here if you have to do it?

It would avoid having mostly the same code twice.

Reply:
See previous remarks and let me know what you prefere